Source/debug.cpp

```diff
@@ -252,6 +253,38 @@ std::string DebugCmdLoadMap(const string_view parameter)
 	return fmt::format("Mapid {} is not known. Do you want to write a mod?", level);
 }
 
+std::string DebugCmdLoadMap(const string_view parameter)
+{
+	ReturnLvlPosition = ViewPosition;
+	ReturnLevel = currlevel;
+	ReturnLevelType = leveltype;
+
+	std::stringstream paramsStream(parameter.data());
+	std::string name;
+	int count = 0;
+	for (std::string tmp; std::getline(paramsStream, tmp, ' '); name += tmp + " ") {
+		switch (count) {
+		case 0:
+			TestMapPath = fmt::format("{:s}.dun", tmp);
+			break;
+		case 1:
+			setlvltype = static_cast<dungeon_type>(atoi(tmp.c_str()));
+			break;
+		case 2:
+			ViewPosition.x = atoi(tmp.c_str());
+			break;
+		case 3:
+			ViewPosition.y = atoi(tmp.c_str());
+			break;
+		}
+		count++;
+	}
+
+	StartNewLvl(MyPlayerId, WM_DIABSETLVL, SL_NONE);

```

Comment:
Should we validate the command and not execute the debug command if the parameters are wrong?
(And don't change the globals (`setlvltype`, `ViewPosition`, ...)?)
For example wrong parameter count, not existing file.
(I know it's only a debug command, only asking 😉)

Reply:
This would be nice, but it was also where I explicitly decided I wasn't going to put more time in to this command :)

Maybe some one else will :sweat_smile:

Reply:
Fine! I threw some basic validation at it :)